applications/h264/grpc_h264_endoscopy_tool_tracking/cpp/edge/app_edge.hpp

```diff
@@ -0,0 +1,66 @@
+/*
+ * SPDX-FileCopyrightText: Copyright (c) 2024 NVIDIA CORPORATION & AFFILIATES. All rights
+ * reserved. SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef GRPC_H264_ENDOSCOPY_TOOL_TRACKING_CPP_APP_EDGE_HPP
+#define GRPC_H264_ENDOSCOPY_TOOL_TRACKING_CPP_APP_EDGE_HPP
+
+#include <holoscan/holoscan.hpp>
+#include <holoscan/operators/format_converter/format_converter.hpp>
+#include <holoscan/operators/holoviz/holoviz.hpp>
+
+#include "video_input_fragment.hpp"
+#include "viz_fragment.hpp"
+
+namespace holohub::grpc_h264_endoscopy_tool_tracking {
+
+using namespace holoscan;
+
+/**
+ * @class AppEdge
+ * @brief A two-fragment application for the H.264 endoscopy tool tracking application.
+ *
+ * This class inherits from the holoscan::Application and is a client application offloads the
+ * inference and process to a remote gRPC server. It is composed with two fragments, a video input
+ * fragment and a visualization fragment using Holoviz.
```

Comment:
What is the advantage of using two fragments here?

Reply:
Updated comments.